sdk/keyvault/azure-keyvault-keys/tests/_shared/test_case.py

```diff
@@ -20,6 +22,16 @@ def setUp(self):
         self.list_test_size = 7
         super(KeyVaultTestCase, self).setUp()
 
+    def create_kv_client(self, vault_uri, **kwargs):
+        credential = self.get_credential(KeyClient)
+        return self.create_client_from_credential(
+            KeyClient, credential=credential, vault_url=vault_uri, **kwargs
```

Comment:
_shared is identical across libraries, so this kind of thing belongs somewhere else. Here's a thought:

```py
class KeyVaultTestCase(AzureTestCase):
    def __init__(self, *args, **kwargs):
        self.client = self.create_client(vault_uri)

class KeyClientTests(KeyVaultTestCase):
    def create_client(self, vault_uri):
        return self.create_client_from_credential(KeyClient, ...)

    # looks like this pattern enables removing the preparer
    def test_something(self):
        self.client.do_something()

        # or, if you don't like touching every client call
        client = self.client
```

Reply:
True, something this would obviate the preparer if I fetch the vault URL before running tests. I actually used a pattern like this for that reason in a prototype, but Laurent would like the SDK to adopt the PowerShellPreparer for common infrastructure and to move away from a dependency on test classes. Moving the client creation out of _shared since the client differs makes sense though

Reply:
What makes the most sense to me is to keep the preparer behavior, but to just move the client creation method into KeyClientTests, etc. What do you think?